mono/eglib/glib.h

```diff
@@ -1386,8 +1386,28 @@ glong     g_utf8_pointer_to_offset (const gchar *str, const gchar *pos);
 #define G_HAVE_API_SUPPORT(x) (x)
 #define G_UNSUPPORTED_API "%s:%d: '%s' not supported.", __FILE__, __LINE__
 #define g_unsupported_api(name) G_STMT_START { g_warning (G_UNSUPPORTED_API, name); } G_STMT_END
- 
-G_END_DECLS
+
+// g_free the result
+// No MAX_PATH limit.
+gboolean
+mono_get_module_filename (gpointer mod, gunichar2** pstr, guint32* plength);
```

Comment:
I can imagine adding `_w` to these function names later.
Or just not calling them on Unix. 
But it is a useful step I think, and layering perhaps.